### PR TITLE
Add support for dotnetbuilds-published dotnet cli packages

### PR DESCRIPTION
**Pull request by alexperovich**
This change adds support for installing dotnet cli packages that were published to dotnetbuilds to both the helix sdk FindDotNetCliPackage task, and the build scripts.

### CODE REVIEW

File: src/Microsoft.DotNet.Helix/Sdk/FindDotNetCliPackage.cs [hunk 1]
```diff
@@ -55,82 +56,114 @@ public override bool Execute()
...
-
-            string downloadUrl = await GetDownloadUrlAsync();
-
-            Log.LogMessage($"Retrieved dotnet cli {PackageType} version {Version} package uri {downloadUrl}, testing...");
-
-            try
+            var feeds = new List<string>
             {
-                using HttpResponseMessage res = await HeadRequestWithRetry(downloadUrl);
+                "https://dotnetcli.azureedge.net/dotnet",
```
Comment (MattGal) [inline src/Microsoft.DotNet.Helix/Sdk/FindDotNetCliPackage.cs:61]: does this mean dotnetcli goes first?  Seems like for a majority of daily builds dotnetbuilds will have the version used.  Probably matters very little except when hitting peak download usage, just curious.

**Review by MattGal — APPROVED** [review]

File: src/Microsoft.DotNet.Helix/Sdk/FindDotNetCliPackage.cs (hunk 1 above)
Comment (alexperovich) [reply src/Microsoft.DotNet.Helix/Sdk/FindDotNetCliPackage.cs:61]: Yea, dotnetcli goes first. My understanding is that we want to prefer the signed bits from dotnetcli over unsigned bits from dotnetbuilds, but if we are okay with our builds using unsigned stuff even if a signed version exists then I can switch the order.

Comment (alexperovich) [reply src/Microsoft.DotNet.Helix/Sdk/FindDotNetCliPackage.cs:61]: it shouldn't really make things much slower, just one server round-trip slower. Its only doing a HEAD request.

Comment (MattGal) [reply src/Microsoft.DotNet.Helix/Sdk/FindDotNetCliPackage.cs:61]: nah that makes sense then, I was thinking we wouldn't have the same version in both places ever so stress the one that CI uses to the betterment of public.

Comment (alexperovich) [reply src/Microsoft.DotNet.Helix/Sdk/FindDotNetCliPackage.cs:61]: Once things are all done, anything new on dotnetcli would have been on dotnetbuilds first, before the release pipeline gets it.

File: eng/common/tools.ps1 [hunk 2]
```diff
@@ -301,31 +301,44 @@ function InstallDotNet([string] $dotnetRoot,
...
+  $variations = @()
+  $variations += @($installParameters)
 
-      if ($runtimeSourceFeedKey) {
-        $decodedBytes = [System.Convert]::FromBase64String($runtimeSourceFeedKey)
-        $decodedString = [System.Text.Encoding]::UTF8.GetString($decodedBytes)
-        $installParameters.FeedCredential = $decodedString
-      }
+  $dotnetBuilds = $installParameters.Clone()
+  $dotnetbuilds.AzureFeed = "https://dotnetbuilds.azureedge.net/public"
```
Comment (mmitche) [inline eng/common/tools.ps1:308]: I didn't think we had a CDN for this account, is this right?

File: eng/common/tools.ps1 [hunk 3]
```diff
@@ -301,31 +301,44 @@ function InstallDotNet([string] $dotnetRoot,
...
+  $variations += @($dotnetBuilds)
 
-      try {
-        & $installScript @installParameters
-      }
-      catch {
-        Write-PipelineTelemetryError -Category 'InitializeToolset' -Message "Failed to install dotnet from custom location '$runtimeSourceFeed'."
-        ExitWithExitCode 1
-      }
+  if ($runtimeSourceFeed) {
```
Comment (mmitche) [inline eng/common/tools.ps1:311]: Are we going to need a variation for dotnetbuilds/internal? Or would we be planning on replacing the existing usage of runtimesourcefeed as dotnetclimsrc in the repos with dotnetbuilds/internal instead?

**Review by mmitche — COMMENTED** [review]
Couple questions, but looks generally okay.

File: eng/common/tools.ps1 (hunk 2 above)
Comment (alexperovich) [reply eng/common/tools.ps1:308]: We do.

File: eng/common/tools.ps1 (hunk 3 above)
Comment (alexperovich) [reply eng/common/tools.ps1:311]: Yea, we should just use dotnetbuilds/internal in place of dotnetclimsrc in our current workflow

**Review by mmitche — APPROVED** [review]

File: eng/common/tools.ps1 (hunk 2 above)
Comment (mmitche) [reply eng/common/tools.ps1:308]: Fancy

**Pull request merged**